flutter-engine/src/texture_registry.rs

```diff
@@ -1,163 +1,88 @@
-use std::{
-    collections::HashMap,
-    convert::TryInto,
-    sync::{atomic::AtomicI64, Arc, Mutex, Weak},
-};
-
-use gl::types::*;
-use log::debug;
-
-use crate::ffi::ExternalTexture as FlutterTexture;
-use crate::ffi::ExternalTextureFrame;
+use crate::ffi::{ExternalTexture, ExternalTextureFrame, TextureId};
 use crate::FlutterEngine;
+use image::RgbaImage;
+use std::collections::HashMap;
+use std::sync::atomic::AtomicI64;
+use std::sync::{Arc, Mutex};
 
-type TextureID = i64;
-
-#[derive(Default)]
+#[derive(Clone, Default)]
 pub struct TextureRegistry {
-    textures: TextureStore,
-}
-
-#[derive(Default)]
-struct TextureStore {
-    /// Stores freshly registered textures without an OpenGL texture attached.
-    initial: HashMap<TextureID, Arc<ExternalTexture>>,
-    /// Stores created textures that may be dropped in other locations.
-    created: HashMap<TextureID, Weak<ExternalTexture>>,
-}
-
-pub struct ExternalTexture {
-    texture: FlutterTexture,
-    texture_data: Mutex<Option<TextureData>>,
-}
-
-struct TextureData {
-    name: GLuint,
-    width: u32,
-    height: u32,
+    textures: Arc<Mutex<HashMap<TextureId, ExternalTexture>>>,
+    data: Arc<Mutex<HashMap<TextureId, RgbaImage>>>,
```

Comment:
Ideally, we'd replace this with just the opengl info. And would save us keeping large image data duplicated in memory many times.

Does an texture being disposed remove it from this map? As otherwise, this would lead to a large memory leak?